Submissions/Bain_Week 3.playground/Contents.swift

```diff
@@ -0,0 +1,227 @@
+//:
+class Shape {
+    var numberOfSides = 0
+    let size = 5
+    func simpleDescription() -> String {
+        return "A shape with \(numberOfSides) sides."
+    }
+    
+    func shapeArea(numberOfSides: Int) -> String {
+        return "The area of your shape is \(size*numberOfSides)."
+    }
+}
+
+//: 5.)
+//:
+//: Make a subclass of `NamedShape` called `Circle` that takes a radius and a name as arguments to its initializer. Implement an `area()` and a `simpleDescription()` method on the `Circle` class.
+//:
+class NamedShape {
+    var numberOfSides: Int = 0
+    var name: String
+    
+    init(name: String) {
+        self.name = name
+    }
+    
+    func simpleDescription() -> String {
+        return "A shape with \(numberOfSides) sides."
+    }
+}
+
+class Square: NamedShape {
+    var sideLength: Double
+    
+    init(sideLength: Double, name: String) {
+        self.sideLength = sideLength
+        super.init(name: name)
+        numberOfSides = 4
+    }
+    
+    func area() -> Double {
+        return sideLength * sideLength
+    }
+    
+    override func simpleDescription() -> String {
+        return "A square with sides of length \(sideLength)."
+    }
+}
+let test = Square(sideLength: 5.2, name: "my test square")
+test.area()
+test.simpleDescription()
+
+class Circle: NamedShape {
+    var radius: Double
+    
+    init(radius: Double, name: String) {
+        self.radius = radius
+        super.init(name: name)
+    }
+    
+    func area() -> Double {
+        return 3.14 * radius * radius
+    }
+    
+    override func simpleDescription() -> String {
+        return "A circle with the radius of \(radius)"
+    }
+}
+
+let test2 = Circle(radius: 4, name: "my test circle")
+test2.area()
+test2.simpleDescription()
+
+//: ## Enumerations and Structures
+//: 6.)
+//:
+//: Write a function that compares two `Rank` values by comparing their raw values.
+//:
+enum Rank: Int {
+    case ace = 1
+    case two, three, four, five, six, seven, eight, nine, ten
+    case jack, queen, king
+    
+    func simpleDescription() -> String {
+        switch self {
+        case .ace:
+            return "ace"
+        case .jack:
+            return "jack"
+        case .queen:
+            return "queen"
+        case .king:
+            return "king"
+        default:
+            return String(self.rawValue)
+        }
+    }
+}
+let ace = Rank.ace
+let aceRawValue = ace.rawValue
+
+//: 7.)
+//:
+//: Add a `color()` method to `Suit` that returns “black” for spades and clubs, and returns “red” for hearts and diamonds.
+//:
+enum Suit: CaseIterable {
+    case spades, hearts, diamonds, clubs
+    
+    func simpleDescription() -> String {
+        switch self {
+        case .spades:
+            return "spades"
+        case .hearts:
+            return "hearts"
+        case .diamonds:
+            return "diamonds"
+        case .clubs:
+            return "clubs"
+        }
+    }
+    
+    func color(suit: Suit) -> String {
```

Comment:
No parameters were asked for; alternatively, you can implement this function by switching on self.

``` Swift
  func color() -> String {
     switch self {
     case .spades, .clubs: 
        return "black"  
     case .hearts, .diamonds:
         return "red"
     }
 }
```

This means you can call the above function with

```Swift
let hearts = Suit.hearts
hearts.color()
```